### PR TITLE
Move copy_from_guest to provision/base.py

### PR DESCRIPTION
**Pull request by t184256**
... and change it a little.

### CODE REVIEW

**Comment by t184256** [comment]
force-push 36c3440: rebase on top of newer, non-broken master

File: tmt/steps/provision/base.py [hunk 1]
```diff
@@ -32,9 +32,25 @@ def copy_from_guest(self, target):
...
-
         """
-        pass
+        beg = f"[[ -d '{target}' ]]"
+        end = 'exit 0; set -xe; '
+
+        isdir = f"{beg} || {end}"
+        isntdir = f"{beg} && {end}"
+
+        def cmd_mkcp(self, tgt_dir, tgt):
```
Comment (pvalena) [inline tmt/steps/provision/base.py]: Please do not make it nested. We've agreed on that with @psss and @thrix; we could possibly move it into common / utils.

Comment (t184256) [reply tmt/steps/provision/base.py]: Un-nested it.

**Comment by t184256** [comment]
There really aren't many changes here:

1. moving `copy_from_guest` across classes unmodified
2. moving `cmd_mkcp` ~~into~~ along with it (changed as per your request)
3. inlining `quote` usage in `cmd_mkcp`:

``` diff
-        target_dir = self.quote(target_dir)
-        target = self.quote(target)
-        return f'mkdir -p {target_dir}; cp -vafr {target} {target_dir}'
+        return f'mkdir -p "{target_dir}"; cp -vafr "{target}" "{target_dir}"'
```

I've expanded the commit message to mention 2 and 3.

**Comment by pvalena** [comment]
>     3. inlining `quote` usage in `cmd_mkcp`:

I don't see why, apart from losing one single-line method...
Maybe there actually is a need for `quote()` into our `utils.py` ... At some point, If we choose to switch to `shlex.quote()`, it will be much simpler. Even at this point, it would be simpler to change / override `quote()` if one needed nesting (the quote could be parametrized; or simply overriden with single-quote).

@psss WDYT?

**Comment by t184256** [comment]
> I don't see why, apart from losing one single-line method...

This, and two lines spent invoking it.

**Comment by pvalena** [comment]
> This, and two lines spent invoking it.

Well, that was only to make it more readable. I really try to avoid hardcoding stuff in favor of reusability. Also prior use of `shlex` was my motivation, and there's a chance we'll need something like that anyway (I've been told `shlex.quote` is in Python 3.7+`, and we need to support 3.6 ATM).

**Comment by t184256** [comment]
Should I revert it and move the quote method as well?

**Comment by pvalena** [comment]
We've agreed \w @thrix to move `quote` into `utils.py`. Can you do it, or do you want me to do it?

_ _ _ _

_Note:_ prepare has it's own variant with single quoting **on purpose** (so it could nest correctly with the vagrant one:
```
$ grep -r 'def quote' -A 2
tmt/steps/prepare/__init__.py:    def quote(self, string):
tmt/steps/prepare/__init__.py-        """ returns string decorated with squot """
tmt/steps/prepare/__init__.py-        return f"'{string}'"
--
tmt/steps/provision/vagrant.py:    def quote(self, string):
tmt/steps/provision/vagrant.py-        """ returns string decorated with dquot """
tmt/steps/provision/vagrant.py-        return f'"{string}"'
```

Please ignore it for now :)

**Comment by t184256** [comment]
Like this?

**Comment by t184256** [comment]
Rebased against current master.

**Comment by psss** [comment]
Looks good. Thanks for moving the method.

**Comment by pvalena** [comment]
> Like this?

Yes thank you both!

_ _ _ _ 
I didn't know you can actually include `quote` what's the behaviour there?

**Pull request merged**